### PR TITLE
Chore: e2e tests on graphic component (Issue/95)

### PR DESCRIPTION
**Pull request by lemmyadams**
Addresses #95

### CODE REVIEW

File: tests/e2e/graphic.cy.js [hunk 1]
```diff
@@ -6,26 +6,38 @@ describe('Graphic', function () {
...
+      let waited = false
+      cy.visit(`/#/preview/${graphicComponent._id}`);
+      const bodyWithoutHtml = graphicComponent.body.replace(/<[^>]*>/g, '');
 
-    this.data.contentObjects.filter((page) => page._classes !== 'assessment').forEach((page) => {
-      cy.visit(`/#/id/${page._id}`);
-      const articlesOnPage = this.data.articles.filter((article) => article._parentId === page._id).map(article => article._id)
-      const blocksOnPage = this.data.blocks.filter((block) => articlesOnPage.includes(block._parentId)).map(blocks => blocks._id)
-      const componentsOnPage = graphicComponents.filter((component) => blocksOnPage.includes(component._parentId))
+      function waitOneSecond() {
```
[the remark this answers is not in the document]
Comment (lemmyadams) [reply tests/e2e/graphic.cy.js:16]: I would have liked to but @cahirodoherty-learningpool mentioned this way of using Promises was better

**Comment by cahirodoherty-learningpool** [comment]
Hi guys, just to clear up a little confusion on this - I think our option here may end up being `cy.wait(X)` as you suggest, but I am hesitant to get into the pattern of throwing waits in too often as it can bloat the running time of the Cypress tests as a whole.
My nudge towards exploring promises/some sort of async manipulation was to avoid that in the cases where possible.

For example, here our `wait` is needed because we are waiting on page load. But it is preferable to react to a known outcome of page load (some event trigger for example) rather than an arbitrary amount of time.

Now I'm not sure of exactly the correct thing to listen out for, but thought it was worth exploring

**Comment by oliverfoster** [comment]
> Hi guys, just to clear up a little confusion on this - I think our option here may end up being `cy.wait(X)` as you suggest, but I am hesitant to get into the pattern of throwing waits in too often as it can bloat the running time of the Cypress tests as a whole. My nudge towards exploring promises/some sort of async manipulation was to avoid that in the cases where possible.
> 
> For example, here our `wait` is needed because we are waiting on page load. But it is preferable to react to a known outcome of page load (some event trigger for example) rather than an arbitrary amount of time.
> 
> Now I'm not sure of exactly the correct thing to listen out for, but thought it was worth exploring

Some interesting strategies here https://stackoverflow.com/questions/64421563/cypress-wait-for-unpredictable-component-to-render-from-an-exclusive-set

File: tests/e2e/graphic.cy.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+describe('Graphic', function () {
+  beforeEach(function () {
+    cy.getData()
+    cy.visit('/');
```
Comment (cahirodoherty-learningpool) [inline tests/e2e/graphic.cy.js:4]: Minor: No need for this visit command as the test forces routing anyway

**Review by cahirodoherty-learningpool — APPROVED** [review]
👀

**Review by ethan-lp — APPROVED** [review]

**Review by joe-allen-89 — APPROVED** [review]
👍

**Pull request merged**